sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -414,6 +414,10 @@ class SQLQuerySuite extends QueryTest with BeforeAndAfterAll {
       sql("with q1 as (select * from testData limit 10) select * from q1"),
       testData.take(10).toSeq)
 
+    intercept[RuntimeException] {
+      sql("with q1 as (select * from testData) with q2 as (select * from q1) select * from q2")
```

Comment:
I see.  It was not clear to me from the description that the goal was to prevent this.  Make this its own test, and call it something like `allow only a single with clause per query`.